index.js

```diff
@@ -31,38 +95,91 @@ module.exports = async (options = {}) => {
 		}
 	}
 
-	const defaultHeaders = [
-		'imageName',
-		'pid',
-		'sessionName',
-		'sessionNumber',
-		'memUsage'
-	];
-
-	const verboseHeaders = defaultHeaders.concat([
-		'status',
-		'username',
-		'cpuTime',
-		'windowTitle'
-	]);
+	let currentHeader;
+	if (options.apps) {
+		currentHeader = 'apps';
+	} else if (options.modules !== undefined) {
+		currentHeader = 'modules';
+	} else if (options.services) {
+		currentHeader = 'services';
+	} else {
+		currentHeader = 'default';
+	}
 
-	const headers = options.verbose ? verboseHeaders : defaultHeaders;
+	if (options.verbose) {
+		currentHeader += 'Verbose';
+	}
 
-	const {stdout} = await execFile('tasklist', args);
+	const columns = csvHeaders[currentHeader];
+	const currentTransform = transforms[currentHeader];
+	return {args, columns, currentTransform};
+}
 
-	// Not start with `"` means no matching tasks. See #11.
-	const data = stdout.startsWith('"') ? await neatCsv(stdout, {headers}) : [];
+/**
+ * Execute tasklist and return results through a stream
+ * @param {Options} options The options of the command
+ * @returns {Stream} Stream, returning parsed results
+ */
+function stream(options = {}) {
```

Comment:
You called the other function `promiseInterface`. Perhaps rename to `streamInterface`? Or remove "interface" from both names?